src/main/java/com/eva/model/person/applicant/ApplicationStatus.java

```diff
@@ -24,7 +22,7 @@ public ApplicationStatus(String status) {
         requireNonNull(status);
         status = status.strip().toLowerCase();
         checkArgument(isValidApplicationStatus(status), MESSAGE_CONSTRAINTS);
-        this.status = status;
+        this.value = PossibleApplicationStatus.getStatus(status);
```

Comment:
Perhaps a better way to process statuses from strings would be to use the `valueOf` function of enums. i.e
`PossibleApplicationStatus.valueOf(status.toUpperCase()); //throws Illegal argument exception if its not a valid enum
`
not really sure if it can be applied here as you would have to try catch either here or whatever is calling the constructor.